core/src/block_creation_loop.rs

```diff
@@ -106,6 +110,23 @@ enum StartLeaderError {
     ),
 }
 
+fn produce_block_footer() -> VersionedBlockMarker {
+    let block_producer_time_nanos = SystemTime::now()
```

Comment:
> `block_producer_time_nanos: u64` is a nanosecond UNIX timestamp representing the time when the block producer started constructing the block. "started
constructing" is the point at which, from the perspective of the leader, all of
the consensus checks required for assuming leadership have "just passed". For
example, in Agave's pre-alpenglow implementations, this would be in
replay/maybe_start_leader. In a post-Alpenglow implementation, this would be
after receiving the proper vote/skip certificate for the previous slot.

So I think what we actually want is 
```rust
    poh_recorder_w.working_bank.unwrap().start
```